packages/headless/src/lib/headless.service.ts

```diff
@@ -690,16 +698,20 @@ export class HeadlessService {
             variables.payload
           ),
         onSuccess: (data) => {
-          this.queryClient.setQueriesData<IMessage[]>(
+          this.queryClient.setQueriesData<IPaginatedResponse<IMessage>>(
             { queryKey: NOTIFICATIONS_QUERY_KEY, exact: false },
-            (oldMessages) =>
-              oldMessages.map((message) => {
-                if (message._id === messageId) {
-                  return data;
-                }
-
-                return message;
-              })
+            (oldMessages) => ({
+              ...oldMessages,
+              ...(oldMessages?.data && {
+                data: oldMessages?.data?.map((message) => {
+                  if (message._id === data._id) {
+                    return data;
+                  }
+
+                  return message;
+                }),
+              }),
+            })
```

Comment:
Great find, but could you please extract the content of `onSucess` in a private function in the class so it is reused in the 3 places we are using it? 🙏🏻

Reply:
Good idea, will do that